src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -31,16 +31,62 @@ public static CodeWriter AppendNullableValue(this CodeWriter writer, CSharpType
             return writer;
         }
 
+        public static IDisposable Method(
```

Comment:
Same comment as for the class helper.

Reply:
I won't agree on this one. It incapsulates 30+ lines of code.

Reply:
it's only 30+ lines because it tries to be generic. It's fewer when you collapse all the ifs.

``` C#
            writer.WriteXmlDocumentationSummary(documentationSummary);
            foreach (var parameter in parameters)
            {
                writer.WriteXmlDocumentationParameter(parameter.Name, parameter.Description);
            }
            writer.WriteXmlDocumentationRequiredParametersException(parameters);
            writer.WriteXmlDocumentationReturns(documentationReturns);

            writer.Append($"{accessibility} {returnType} {name}(");
            foreach (var parameter in parameters)
            {
                writer.WriteParameter(parameter);
            }
            writer.RemoveTrailingComma();
            writer.Append($")");

            return writer.Scope();
```

The bigger point is that while having this helper in isolation might be a good thing there are many developers in the codebase that don't know of its existence and won't use it. If we think this is the right direction to go we should do it in a separate PR and update all the usages.

Reply:
Resolved